xds/src/test/java/io/grpc/xds/XdsClientMetricReporterImplTest.java

```diff
@@ -139,16 +140,19 @@ public void setXdsClient_reportCallbackMetrics_resourceCountsFails() {
 
   @Test
   public void metricGauges() {
+    XdsClientMetricReporterImpl spyReporter = spy(
```

Comment:
Never, ever use a spy. The way they are implemented essentially invalidates any testing using them. Use a mock with `delegatesTo(realImpl)`.

Reply:
When I was modifying the test to remove spy, realized I do not need `delegatesTo` either.